docs/content/topics/operators/gateways.md

```diff
@@ -11,340 +11,421 @@ aliases:
+[Javascript/Typescript SDK]: https://github.com/brigadecore/brigade-sdk-for-js
+[Rust SDK]: https://github.com/brigadecore/brigade-sdk-for-rust
 
-## Creating A Cron Job Gateway
+## Example Gateway
 
-Beginning with the code above, we can build a gateway that runs as a scheduled
-job in Kubernetes. In this example, we use a Kubernetes
-[CronJob](https://kubernetes.io/docs/concepts/workloads/controllers/cron-jobs/) object
-to create the secret.
+The following example assumes a running Brigade instance has been deployed and
+the ability to create a service account is in place (e.g. you have the role of
+'ADMIN' or you are logged in as the root user). If you'd like to follow along
+and haven't yet deployed Brigade, check out the [QuickStart].
 
-First we can begin with a simplified version of the script above. This one does not
-run in a loop. It just runs once to completion.
+[QuickStart]: /intro/quickstart
 
-Here is `cron-event.sh`:
+### Preparation
 
-```bash
-#!/usr/bin/env bash
-set -euo pipefail
+#### Service Account creation
 
-# The Kubernetes namespace in which Brigade is running.
-namespace=${NAMESPACE:-default}
+All Brigade gateways require a service account token for authenticating with
+Brigade when submitting an event into the system. As preparation then, we'll
+create a service account for this gateway and save the generated token for use
+in our program.
 
-event_provider="simple-event"
-event_type="my_event"
-project_id="brigade-830c16d4aaf6f5490937ad719afd8490a5bcbef064d397411043ac"
-commit_ref="master"
-commit_id="589e15029e1e44dee48de4800daf1f78e64287c0"
-uuid="$(uuidgen | tr '[:upper:]' '[:lower:]')"
-name="simple-event-$uuid"
+```console
+$ brig service-account create \
+	--id example-gateway \
+	--description example-gateway
+```
 
-payload=$(date)
-script=$(cat <<EOF
-const { events } = require("brigadier");
-events.on("my_event", (e) => {
-  console.log("The system time is " + e.payload);
-});
-EOF
-)
+Make note of the token returned. This value will be used in another step. It is
+your only opportunity to access this value, as Brigade does not save it.
 
-cat <<EOF | kubectl --namespace ${namespace} create -f -
-apiVersion: v1
-kind: Secret
-metadata:
-  name: ${name}
-  labels:
-    heritage: brigade
-    project: ${project_id}
-    build: ${uuid}
-    component: build
-type: "brigade.sh/build"
-data:
-  revision:
-    commit: $(base64 -w 0 <<<"${commit_id}")
-    ref: $(base64 -w 0 <<<"${commit_ref}")
-  event_provider: $(base64 -w 0 <<<"${event_provider}")
-  event_type: $(base64 -w 0 <<<"${event_type}")
-  project_id: $(base64 -w 0 <<<"${project_id}")
-  build_id: $(base64 -w 0 <<<"${uuid}")
-  payload: $(base64 -w 0 <<<"${payload}")
-  script: $(base64 -w 0 <<<"${script}")
-EOF
+Authorize this service account to create new events:
+
+```console
+$ brig role grant EVENT_CREATOR \
+    --service-account example-gateway \
+    --source brigade.sh/example-gateway
```

Comment:
The rule of thumb to avoid source clashes is to use a URI you control. This means leading with one's own domain or the URL for something else one owns, like the URL for a github repo, for instance.

Reply:
👍 So I take it we'd like to add a mention here around the naming convention of the source value?

Reply:
We probably should, but at the very least, we probably shouldn't be instructing people to use _our_ domain name as a component of their source string.

Reply:
Oh gotcha - yup.

Reply:
Note added (and example uri updated).